betty/requirement.py

```diff
@@ -238,10 +245,58 @@ def requires_project(
     async def _requires_project(
         cls: type[_HasRequirementT], service_level: ServiceLevel
     ) -> Requirement | None:
-        from betty.project import Project
-
         if not isinstance(service_level, Project):
             return StaticRequirement(Plain("A Project is required."))
         return await f(cls, service_level)
 
     return _requires_project
+
+
+def requires_extension(
+    extension: ResolvableId[ExtensionDefinition, Extension],
+) -> Callable[
+    [
+        Callable[
+            [type[_HasRequirementT], _ExtensionT],
+            Coroutine[Any, Any, Requirement | None],
+        ]
+    ],
+    Callable[
+        [type[_HasRequirementT], ServiceLevel], Coroutine[Any, Any, Requirement | None]
+    ],
+]:
+    """
+    Decorate a :py:meth:`betty.requirement.HasRequirement.requirement` implementation to require an :py:class:`betty.project.extension.Extension`.
+    """
+
+    def _requires_extension(
+        f: Callable[
+            [type[_HasRequirementT], _ExtensionT],
+            Coroutine[Any, Any, Requirement | None],
+        ],
+    ) -> Callable[
+        [type[_HasRequirementT], ServiceLevel], Coroutine[Any, Any, Requirement | None]
+    ]:
+        async def __requires_extension(
+            cls: type[_HasRequirementT], service_level: ServiceLevel
+        ) -> Requirement | None:
+            nonlocal extension
+
+            extension = resolve_id(extension)
+            if (
+                not isinstance(service_level, Project)
+                or extension not in await service_level.extensions
+            ):
+                return StaticRequirement(
+                    Plain(
+                        "A Project with the {extension} extension is required."
```

Comment:
For all three decorators, localize the message and write it for a primary audience who work with a single project through the console.